### PR TITLE
Add analytics

### PR DESCRIPTION
**Pull request by chr1sdi**
https://trello.com/c/seTp90X4/421-add-support-for-google-analytics

### CODE REVIEW

File: src/test/java/uk/gov/register/functional/AnalyticsFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        Boolean docIncludesAnalyticsId = trackingIdElem != null;
+        Boolean docIncludesMainAnalytics = doc.getElementById("analytics-main") != null;
+        Boolean docIncludesExtLinksAnalytics = doc.getElementById("analytics-external-links") != null;
+
+        assertThat(docIncludesAnalyticsId, equalTo(shouldIncludeAnalytics));
+        assertThat(docIncludesMainAnalytics, equalTo(shouldIncludeAnalytics));
+        assertThat(docIncludesExtLinksAnalytics, equalTo(shouldIncludeAnalytics));
+
+        if (shouldIncludeAnalytics) {
+            assertThat(docIncludesAnalyticsId, equalTo(true));
```
Comment (karlbaker02) [inline src/test/java/uk/gov/register/functional/AnalyticsFunctionalTest.java]: This is a duplicate of line 82: if `shouldIncludeAnalytics` is `true` before then we're going to run the same check again here

File: src/main/resources/assets/js/analytics-download-type.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+(function(){
+    var elems = document.getElementsByClassName("js-download");
+
+    setupEventAnalytics(elems, "Data", "download", function(e){
+        return e.target.getAttribute("data-download-type") || "unrecognised";
+    });
+
+    setupEventAnalytics(elems, "Data", "download", function(e){
```
Comment (karlbaker02) [inline src/main/resources/assets/js/analytics-download-type.js]: Is this second function intended? Same signature, same name.

File: src/test/java/uk/gov/register/functional/AnalyticsFunctionalTest.java (hunk 1 above)
Comment (chr1sdi) [inline src/test/java/uk/gov/register/functional/AnalyticsFunctionalTest.java]: was added for clarity of intention, will remove it if the test is clear without it

File: src/main/resources/assets/js/analytics-download-type.js (hunk 2 above)
Comment (chr1sdi) [inline src/main/resources/assets/js/analytics-download-type.js]: it's an invocation of the setup for 2nd event on data download click

**Pull request merged**